Foxtrot/Driver/Rewriting/Utils/TypeNodeExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Compiler;
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.Contracts.Foxtrot.Utils
+{
+    internal static class TypeNodeExtensions
+    {
+        public static int TemplateArgumentsCount(this TypeNode typeNode)
+        {
+            Contract.Requires(typeNode != null);
+
+            return typeNode.TemplateArguments == null ? 0 : typeNode.TemplateArguments.Count;
+        }
+
+        public static int CountOrDefault(this TypeNodeList list)
```

Comment:
:thought_balloon: I don't like this extension method. Soon we will be able to use the following, and until then I think we should just write the expression where it's used.

```
list?.Count();
```